executor/insert_common.go

```diff
@@ -224,9 +236,121 @@ func insertRows(ctx context.Context, base insertCommon) (err error) {
 			}
 		}
 	}
+	// Fill the batch allocated autoIDs
+	rows, err = e.autoIDAllocN(ctx, rows)
+	if err != nil {
+		return err
+	}
 	return base.exec(ctx, rows)
 }
 
+func (e *InsertValues) autoIDAllocN(ctx context.Context, rows [][]types.Datum) ([][]types.Datum, error) {
+	retryInfo := e.ctx.GetSessionVars().RetryInfo
+	length := len(e.cache)
+	for i := 0; i < length; i++ {
+		switch e.cache[i].kind {
+		case AutoIDNull:
+			// Find consecutive num.
+			var cnt int
+			var start int
```

Comment:
```suggestion
			var start int
```
Simply `start := i` is enough

Reply:
Cool Done!  >.<